xbmc/powermanagement/PowerManager.h

```diff
@@ -95,6 +96,8 @@ class CPowerManager : public IPowerEventsCallback
   void OnLowBattery() override;
 
   IPowerSyscall *m_instance;
+  CFileItemPtr m_lastFileItem;
```

Comment:
I am no a fan of those crappy typedefs that require an include some hack workaround for a forward declaration.

better:
class CFileItem;

std::shared_ptr\<CFileItem> m_lastFileItem;

Reply:
I'm with you, I don't know any advantages on simple or short type definitions either.